### PR TITLE
feat: docker/docker-compose support

### PR DESCRIPTION
**Pull request by vvisionnn**
- support docker deploy with three different architectures (`arm[v7/64]` and `amd64`)
- `docker-compose` support with remote download image integrated
- github actions automatically release docker image on every push with release tag (3.* currently)

### CODE REVIEW

**Comment by topjohncian** [comment]
Aria2 是否可以成为可选项

**Comment by vvisionnn** [comment]
> Aria2 是否可以成为可选项

I think the native docker version is the optional
you can choose
- use offline download => choose docker-compose
- not => native docker container
no conflict on both of them

thanks

**Review by HFO4 — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,70 +1,43 @@
...
+    && tag_name=$(git describe --tags) \
     && export COMMIT_SHA=$(git rev-parse --short HEAD) \
-    && export VERSION=$(git describe --tags) \
-    && go install -ldflags "-X 'github.com/cloudreve/Cloudreve/v3/pkg/conf.BackendVersion=${VERSION}' \
-                            -X 'github.com/cloudreve/Cloudreve/v3/pkg/conf.LastCommit=${COMMIT_SHA}'\
-                            -w -s"
-
-# build final image
-FROM alpine:3.12 AS dist
+    && go build -a -o cloudreve -ldflags " -X 'github.com/HFO4/cloudreve/pkg/conf.BackendVersion=$tag_name' -X 'github.com/HFO4/cloudreve/pkg/conf.LastCommit=$COMMIT_SHA'"
```
Comment (XYenon) [inline Dockerfile:21]: Should be cloudreve/Cloudreve

Comment (vvisionnn) [reply Dockerfile:21]: Good catch! I'll create another PR for fixing this, thanks!

**Pull request merged**